lib/driver.rb

```diff
@@ -0,0 +1,99 @@
+require_relative 'user'
+require 'pry'
+
+module RideShare
+  class Driver < User
+    attr_reader :vin, :driven_trips, :status
+
+    def initialize(input)
+      super
+
+      @vin = input[:vin]
+      if @vin.length != 17
+        raise ArgumentError.new("Invalid VIN")
+      else
+        @vin = input[:vin]
+      end
+
+      @driven_trips = input[:driven_trips].nil? ? [] : input[:driven_trips]
+
+      if input[:status] != :AVAILABLE && input[:status] != :UNAVAILABLE
+        @status = :UNAVAILABLE
+      else
+        @status = input[:status]
+      end
+    end
+
+    def add_driven_trip(trip)
+      if trip.is_a? Trip
+      # binding.pry
+        @driven_trips << trip
+      # binding.pry
+      else
+        raise ArgumentError.new("Invalid trip instance")
+      end
+    end
+
+    def average_rating
+      total_num_rides = @driven_trips.length
+      total_sum = 0
+
+      if total_num_rides == 0
+        return 0
+      end
+
+      @driven_trips.each do |x|
+        # binding.pry
+        if x.rating == "In Progress"
+          next
```

Comment:
You tended to use `.each` iteration to calculate totals/sum things in Driver and User, even though things like `.sum` is available. Honestly, I don't believe that you ALWAYS need to use fancy Enumerable methods like `.sum` over `.each` iteration every time, but I want to make sure you know that those are possibilities.